lib/ts/BaseLogFile.cc

```diff
@@ -0,0 +1,571 @@
+/** @file
+
+  Base class for log files implementation
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include "BaseLogFile.h"
+
+/*
+ * This consturctor creates a BaseLogFile based on a given name.
+ * This is the most common way BaseLogFiles are created.
+ */
+BaseLogFile::BaseLogFile(const char *name)
+  : m_name(ats_strdup(name)), m_hostname(NULL), m_is_regfile(false), m_has_signature(false), m_signature(0), m_is_init(false)
+{
+  m_fp = NULL;
+  m_start_time = time(0);
+  m_end_time = 0L;
+  m_bytes_written = 0;
+  m_meta_info = NULL;
+
+  log_log_trace("exiting BaseLogFile constructor, m_name=%s, this=%p\n", m_name, this);
+}
+
+/*
+ * This consturctor creates a BaseLogFile based on a given name.
+ * Similar to above constructor, but is overloaded with the object signature
+ */
+BaseLogFile::BaseLogFile(const char *name, uint64_t sig)
+  : m_name(ats_strdup(name)), m_hostname(NULL), m_is_regfile(false), m_has_signature(true), m_signature(sig), m_is_init(false)
+{
+  m_fp = NULL;
+  m_start_time = time(0);
+  m_end_time = 0L;
+  m_bytes_written = 0;
+  m_meta_info = NULL;
+
+  log_log_trace("exiting BaseLogFile constructor, m_name=%s, this=%p\n", m_name, this);
+}
+
+/*
+ * This copy constructor creates a BaseLogFile based on a given copy.
+ */
+BaseLogFile::BaseLogFile(const BaseLogFile &copy)
+  : m_fp(NULL), m_start_time(copy.m_start_time), m_end_time(0L), m_bytes_written(0), m_name(ats_strdup(copy.m_name)),
+    m_hostname(ats_strdup(m_hostname)), m_is_regfile(false), m_has_signature(copy.m_has_signature), m_signature(copy.m_signature),
+    m_is_init(copy.m_is_init), m_meta_info(NULL)
+{
+  log_log_trace("exiting BaseLogFile copy constructor, m_name=%s, this=%p\n", m_name, this);
+}
+
+/*
+ * Destructor.
+ */
+BaseLogFile::~BaseLogFile()
+{
+  log_log_trace("entering BaseLogFile destructor, m_name=%s, this=%p\n", m_name, this);
+
+  if (m_is_regfile)
+    close_file();
+  else
+    log_log_trace("not a regular file, not closing, m_name=%s, this=%p\n", m_name, this);
+  if (m_name)
+    ats_free(m_name);
```

Comment:
I took a look and (I might be wrong) it looks like `ats_scoped_str` 's only live as long as the most immediate scope. I'm not sure how it could be used for the life of an object.